ctpolicy/loglist/loglist.go

```diff
@@ -0,0 +1,164 @@
+package loglist
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/letsencrypt/boulder/ctpolicy/ctconfig"
+	"github.com/letsencrypt/boulder/ctpolicy/loglist/generated"
+)
+
+// logList is a cached/memoized in-memory version of the autogenerated and
+// embedded JSON blob. We do this so we don't have to re-parse the whole blob
+// every time we want a new log list.
+var logList struct {
+	sync.Mutex
+	ready bool
+	generated.LogListSchemaJson
+}
+
+// LogList represents a list of logs, grouped by their operator, arranged by
+// the "v3" schema as published by Chrome:
+// https://www.gstatic.com/ct/log_list/v3/log_list_schema.json
+// It exports no fields so that consumers don't have to deal with the terrible
+// autogenerated names of the structs it wraps.
+type LogList map[string]operatorGroup
+
+// operatorGroup is a convenience type allowing constant-time lookup of logs
+// within a group by their unique ID.
+type operatorGroup map[string]*generated.LogListSchemaJsonOperatorsElemLogsElem
+
+// New returns a LogList built from the autogenerated and embedded JSON blob.
+func New() (LogList, error) {
+	if !logList.ready {
+		logList.Lock()
+		defer logList.Unlock()
+
+		err := json.Unmarshal(generated.LogListJSON, &logList)
+		if err != nil {
+			return nil, err
+		}
+
+		logList.ready = true
+	}
+
+	result := make(LogList)
+	for _, op := range logList.Operators {
+		group := make(operatorGroup)
+		for i := range op.Logs {
+			log := op.Logs[i]
+			group[log.LogId] = &log
+		}
+		result[op.Name] = group
+	}
+
+	return result, nil
+}
+
+// NewFromLogIDs returns a LogList which is a subset of the autogenerated and
+// embedded JSON blob, containing only those logs whose LogIDs match the given
+// list of IDs.
+func NewFromLogIDs(ids []ctconfig.LogID) (LogList, error) {
+	if !logList.ready {
+		logList.Lock()
+		defer logList.Unlock()
+
+		err := json.Unmarshal(generated.LogListJSON, &logList)
+		if err != nil {
+			return nil, err
+		}
+
+		logList.ready = true
+	}
+
+	fmt.Printf("Loading log list with IDs: %v\n", ids)
+
+	result := make(LogList)
+	for _, op := range logList.Operators {
+		group := make(operatorGroup)
+		for i := range op.Logs {
+			log := op.Logs[i]
```

Comment:
Actually did end up doing it this way, due to the other change which means we're building a map of new objects, rather than just pointing to existing ones, so we're no longer taking the address of a loop variable.